src/main/groovy/life/qbic/DownloadRequest.groovy

```diff
@@ -13,75 +15,27 @@ import ch.ethz.sis.openbis.generic.dssapi.v3.dto.datasetfile.DataSetFile
  * @since 0.4.0
  */
 class DownloadRequest {
+  private final List<DataSetFile> dataSetFiles;
+  private final Path prefix
+  private final int retries
 
-  final private Map<String, DataSetFile> dataSetByPermId
-
-  private String sampleCode
-
-  private int retries
-
-  DownloadRequest() {
-    this.dataSetByPermId = new HashMap<>()
-    this.sampleCode = ""
-    this.retries = 1
-  }
-
-  /**
-   * Download request constructor with a provided list of data set files.
-   *
-   * @param dataSetFiles
-   */
-  DownloadRequest(List<DataSetFile> dataSetFiles, String sampleCode) {
-    this()
-    this.sampleCode = sampleCode
-    dataSetFiles.each { dsFile ->
-      addDataSet(dsFile.getPermId().toString(), dsFile)
-    }
-    this.retries = 1
-  }
 
   /**
    * Download request constructor with a provided list of data set files and a configured
    * number of retries on failure.
    *
-   * @param dataSetFiles
-   * @param sampleCode
+   * @param datasetSampleCode the code of the sample to which the dataset was attached
+   * @param dataSetFiles the files to download
    * @param numberRetries The number of retries. Must be >=1, else it will be set to 1
    */
-  DownloadRequest(List<DataSetFile> dataSetFiles, String sampleCode, int numberRetries) {
-    this()
-    this.sampleCode = sampleCode
-    dataSetFiles.each { dsFile ->
-      addDataSet(dsFile.getPermId().toString(), dsFile)
-    }
-    this.retries = numberRetries >= 1 ? numberRetries : 1
-  }
+  DownloadRequest(Path prefix, List<DataSetFile> dataSetFiles, int numberRetries) {
```

Comment:
prefix not explained - I assume this is the dataset code and has nothing to do with the sample?

Reply:
No this is the path to where the download should be downloaded. The prefix for the file (e.g. `/home/users/..../QABCDDATA/`)

Reply:
Ah, that makes sense! Currently this change is not reflected in the Javadoc